### PR TITLE
Updated virtual media resource

### PR DESCRIPTION
**Pull request by prajwalpatil25**
Updated virtual media resource:
- Added support for iDRAC firmware version 6.x and 7.x
- Backward compatibility with version 5.x
- Removed dependency on virtual media ID. Image is inserted on available/free slot.
- For version 5.x, virtual media ID is determined depending on the type of image provided.

image

### CODE REVIEW

File: redfish/resource_redfish_virtual_media.go [hunk 1]
```diff
@@ -197,13 +175,70 @@ func createRedfishVirtualMedia(service *gofish.Service, d *schema.ResourceData)
...
+	//Get Systems details
+	systems, err := service.Systems()
 	if err != nil {
-		return diag.Errorf("Couldn't mount Virtual Media: %s", err)
+		return diag.Errorf("Error when retrieving systems: %s", err)
 	}
 
-	d.SetId(virtualMedia.ODataID)
-	return diags
+	virtualMediaCollection, err := systems[0].VirtualMedia()
```
Comment (rounak-adhikary) [inline redfish/resource_redfish_virtual_media.go:184]: Please include a check to see if zero systems have been returned by API.

Comment (prajwalpatil25) [reply redfish/resource_redfish_virtual_media.go:187]: added.

**Review by rounak-adhikary — COMMENTED** [review]
LGTM

**Review by rounak-adhikary — APPROVED** [review]

**Review by rounak-adhikary — DISMISSED** [review]
LGTM

**Review by shenda1 — DISMISSED** [review]

File: redfish/resource_redfish_dell_idrac_attributes.go [hunk 2]
```diff
@@ -231,7 +231,7 @@ func getManagerAttributeRegistry(service *gofish.Service) (*dell.ManagerAttribut
 	for _, r := range registries {
 		if r.ID == "ManagerAttributeRegistry" {
 			// Get ManagerAttributesRegistry
-			managerAttrRegistry, err := dell.GetDellManagerAttributeRegistry(service.Client, r.Location[0].URI)
+			managerAttrRegistry, err := dell.GetDellManagerAttributeRegistry(service.GetClient(), r.Location[0].URI)
```
Comment (shenda1) [inline redfish/resource_redfish_dell_idrac_attributes.go:234]: Are we sure r.Location[0].URI will not result in nil reference?

Comment (prajwalpatil25) [reply redfish/resource_redfish_dell_idrac_attributes.go:234]: GetClient() changes are to adopt latest gofish version. These changes are not related to virtual media resource implementation.

**Review by shenda1 — APPROVED** [review]

**Review by rounak-adhikary — APPROVED** [review]

**Pull request merged**